### ISSUE TITLE
:package fails when no signing identity is specified

### ISSUE THREAD

**Issue by pjorquera** [openbakery/gradle-xcodePlugin#149]
When building the provided SwiftExample using `gradle release` the process fails with the following output:

```
Configure sign settings for the integration and release build
:xcodebuildConfig
:infoplistModify
Updating /Users/pjorquera/Developer/gradle-xcodePlugin/example/iOS/SwiftExample/SwiftExample/Info.plist
:keychainClean
:keychainCreate
:provisioningClean
:provisioningInstall
No provisioning profile specifed so do nothing here
:xcodebuild
      OK - Linking: build/obj/SwiftExample.build/Debug-iphoneos/SwiftExample.build/Objects-normal/armv7/SwiftExample
      OK - Linking: build/obj/SwiftExample.build/Debug-iphoneos/SwiftExample.build/Objects-normal/arm64/SwiftExample
      OK - Compile: SwiftExample/Base.lproj/LaunchScreen.xib
      OK - Compile: SwiftExample/Base.lproj/Main.storyboard
Done
:archive
convert plist to binary /Users/pjorquera/Developer/gradle-xcodePlugin/example/iOS/SwiftExample/build/archive/SwiftExample.xcarchive/Products/Applications/SwiftExample.app/Base.lproj/Main.storyboardc/Info.plist
convert plist to binary /Users/pjorquera/Developer/gradle-xcodePlugin/example/iOS/SwiftExample/build/archive/SwiftExample.xcarchive/Products/Applications/SwiftExample.app/Info.plist
:package
Set Info Plist Value: Delete CFBundleResourceSpecification
:package FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':package'.
> Ambiguous method overloading for method org.gradle.api.logging.Logging$LoggerImpl#lifecycle.
  Cannot resolve which method to invoke for [class java.lang.String, null] due to overlapping prototypes between:
  	[class java.lang.String, class [Ljava.lang.Object;]
  	[class java.lang.String, class java.lang.Throwable]
```

I removed the section `signing` from `integration` and `release` task configuration (in build.gradle) so I expected that the build will use my `project.pbxproj` default signing values (currently I configured my signing identity in `project.pbxproj` instead `build.gradle` or `gradle.properties`):

```
if (taskGraph.hasTask(integration) || taskGraph.hasTask(release)) {
    println "Configure sign settings for the integration and release build"
    xcodebuild {
        sdk = 'iphoneos'
    }
}
```

Maybe the problem is in line 144 of PackageTask.groovy: https://github.com/openbakery/gradle-xcodePlugin/blob/e5b3e430fbd244eb4d18758d92d3f6c8fc0d7146/plugin/src/main/groovy/org/openbakery/signing/PackageTask.groovy#L144

Probably if identity is `null` in this line:

```
logger.lifecycle("Codesign with Identity: {}", project.xcodebuild.getSigning().getIdentity())
```

the build execution will be:

```
> Ambiguous method overloading for method org.gradle.api.logging.Logging$LoggerImpl#lifecycle.
  Cannot resolve which method to invoke for [class java.lang.String, null] due to overlapping prototypes between:
  	[class java.lang.String, class [Ljava.lang.Object;]
  	[class java.lang.String, class java.lang.Throwable]
```

**Comment by renep** [comment]
release build means you want to create an ipa that can be deployed on the device. This does NOT work without signing. Therefor the signing is needed and you need to specify your sign identity, key and provisioning profile.

If you only test if the project compiles than run `gradle xcodebuild` or `gradle continuous`

**Comment by pjorquera** [comment]
Thank you very much René. I know that `release` builds needs signing configuration, but I would like to not configure that using `build.gradle` or `gradle.properties`.

Is it possible to specify that signing identity must be used from what is configured in `project.pbxproj` instead `build.gradle` or `gradle.properties`?

In previous release of the plugin (0.9.15) I could successfully sign IPAs (using `gradle archive` of course, not `gradle package`) without needing to configure code signing identity (neither key or provisioning profile) because this signing configuration was retrieved from what is specified in `project.pbxproj`.

Anyway, thank you very much for developing this plugin. Currently we are using v0.9.15 for building all our apps using your plugin and sending IPAs to Crashlytics Beta using post custom Gradle tasks.

**Comment by renep** [comment]
You can use the keychain parameter at the signing and specify your login keychain. e.g. 

```
signing {
 keychain="~/Library/Keychains/login.keychain"
}
```

**Comment by pjorquera** [comment]
Now I can successfully code sign specifying keychain, but only if I use *full path* to keychain file. Relative paths are created if I use HOME directory `~`:

```
signing {
    keychain = "~/Library/Keychains/login.keychain"
}
```

results in the following command:

```
/usr/bin/codesign --force --sign "iPhone Distribution" --verbose
/Users/pjorquera/demo/build/codesign/Payload/Demo.app/Frameworks/libswiftCore.dylib
--keychain /Users/pjorquera/demo/~/Library/Keychains/login.keychain
```

instead the following (expected):
```
/usr/bin/codesign --force --sign "iPhone Distribution" --verbose
/Users/pjorquera/demo/build/codesign/Payload/Demo.app/Frameworks/libswiftCore.dylib
--keychain ~/Library/Keychains/login.keychain
```

**Comment by renep** [comment]
Then use:

```
signing {
    keychain = "/Users/pjorquera/demo/Library/Keychains/login.keychain"
}
```

**Comment by renep** [comment]
I have created a new issue for the problem #150

**Comment by pjorquera** [comment]
That's not suitable in our environment (Jenkins CI). Will wait until a fix for #150.

**Comment by renep** [comment]
If you want to do it right, than do NOT use the keychain parameter. 
Why? If you have a team with multiple developers than every one can change the settings in the Xcode project file, and this will break the build.
So the idea behind specifying the sign settings in the gradle file is that it does overrule the settings in the project file.
In my setup the keys and provisioning profiles are stored in one place in my LAN, and I only have to update it in a single location is something changes. This helps a lot, because I have multiple build nodes and I do not have to updated the keychain on every build node if something changes.
